langchain4j-github-models/pom.xml

```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>dev.langchain4j</groupId>
+        <artifactId>langchain4j-parent</artifactId>
+        <version>0.34.0</version>
+        <relativePath>../langchain4j-parent/pom.xml</relativePath>
+    </parent>
+
+    <artifactId>langchain4j-github-models</artifactId>
+    <name>LangChain4j :: Integration :: GitHub Models</name>
+
+    <dependencies>
+
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j-core</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>com.azure</groupId>
+            <artifactId>azure-ai-inference</artifactId>
```

Comment:
I thought GitHub models exposed openai apis - can't we reuse existing api ? Or is there a value add of adding  such dependency?

Reply:
Have a look at https://github.com/marketplace/models/azure-openai/gpt-4o in "get started", and check the API in JavaScript (Java will be added soon): you have the choice between OpenAI and the Azure AI Inference API.
So yes, the OpenAI models should work for accessing OpenAI models.
Then, for https://github.com/marketplace/models/azureml-mistral/Mistral-Nemo you have the choice between the Mistral AI SDK and the Azure AI Inference API.
And for https://github.com/marketplace/models/azureml/Phi-3-5-mini-instruct it will be only the Azure AI Inference API.
So the Azure AI Inference API is the only one which should work across all models. I agree, differences are probably very light, but that might change.

Reply:
In "Get Started," if we check the API specifications for REST—whether it's "gpt-4o," "Mistral-Nemo," or "Phi-3.5-mini-instruct"—the API specifications are the same for all and are compatible with OpenAI's API, which uses a Bearer token. Therefore, if you use langchain4j's OpenAiChatModel API, it will work with all the supported GitHub models. The same code snippet I mentioned in issue(https://github.com/langchain4j/langchain4j/issues/1719#issuecomment-2361872630)  works for all the GitHub models you referenced.

Now, regarding the "Azure AI Inference API," you’re correct that it may not currently support Java. However, langchain4j's OpenAiChatModel API alone suffices for your requirements to use GitHub models. Internally, it does not rely on any SDKs like the Mistral AI SDK or the Azure AI Inference SDK.

Aside from this, is there any specific requirement that langchain4j's OpenAiChatModel API cannot address?

Reply:
Oh yes: it’s not done by the companies doing those models, so you can’t rely on them for maintenance, support, new features… Most companies will refuse to use the non official SDK, and I can confirm this on the Azure OpenAI side where I have the LangChain4J OpenAI and the LangChain4J Azure OpenAI stats, which is exactly the same situation - nearly everyone uses the official SDK.
The Java version of the Azure AI Inference API had its first release yesterday.

Reply:
I will also have access to the GitHub Models stats, so I’ll be able to tell which implementation people use.
If this implementation isn’t popular, you can trust me that I will remove this module.

Reply:
> Oh yes: it’s not done by the companies doing those models, so you can’t rely on them for maintenance, support, new features… Most companies will refuse to use the non official SDK, and I can confirm this on the Azure OpenAI side where I have the LangChain4J OpenAI and the LangChain4J Azure OpenAI stats, which is exactly the same situation - nearly everyone uses the official SDK.
> The Java version of the Azure AI Inference API had its first release yesterday.

I agree, we should use the official SDK.